ui/pages/confirmations/hooks/useCurrentConfirmation.ts

```diff
@@ -59,16 +70,20 @@ const useCurrentConfirmation = () => {
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     (signatureMessage?.msgParams as any)?.siwe?.isSIWEMessage;
 
+  // This makes sure that only the user setting is enabled to show the redesigned signatures
+  const isSignatureRedesignRequested =
+    isRedesignedConfirmationsUserSettingEnabled && isCorrectApprovalType;
+  // This makes sure that the developer setting && user setting are enabled to show the redesigned transaction
+  const isRedesignedTransactionRequested =
+    isRedesignedConfirmationsUserSettingEnabled &&
+    isRedesignedConfirmationsDeveloperSettingEnabled &&
+    isCorrectTransactionType;
+  const shouldUseRedesign =
+    !(isSignatureRedesignRequested || isRedesignedTransactionRequested) ||
+    isSIWE;
+
   return useMemo(() => {
-    if (
-      !redesignedConfirmationsEnabled ||
-      (!isCorrectTransactionType && !isCorrectApprovalType) ||
-      /**
-       * @todo remove isSIWE check when we want to enable SIWE in redesigned confirmations
-       * @see {@link https://github.com/MetaMask/metamask-extension/issues/24617}
-       */
-      isSIWE
-    ) {
+    if (shouldUseRedesign) {
```

Comment:
The name of this is inverted so currently we're using it as if it was `shouldNotUseRedesign` but I've suggested the positive condition in another comment which I think is clearer.

Reply:
done